### PR TITLE
feat: IXamlMethod Generic definition

### LINKED ISSUE #112: IXamlMethod Generic definition
HI,
in the pr https://github.com/AvaloniaUI/Avalonia/pull/15274. I have need to retrieve `Interactive.AddHandler<TEventArgs>` method. To do this, I tried using `IXamlType.FindMethod` and checking if the method name is **AddHandler`1** without success.

After investigation I found following workaround

https://github.com/AvaloniaUI/Avalonia/blob/3f82e64c469198d3e80557b7c3c5071a8c29d806/src/Markup/Avalonia.Markup.Xaml.Loader/CompilerExtensions/Transformers/AvaloniaXamlIlWellKnownTypes.cs#L155-L163

```c#
                AddHandlerT = Interactive.FindMethod(m => m.IsPublic
                    && !m.IsStatic
                    && m.Name == "AddHandler"
                    && m.Parameters.Count == 4
                    && RoutedEvent.IsAssignableFrom(m.Parameters[0])
                    && m.Parameters[0].GenericArguments?.Count == 1 // This is specific this case  workaround to check is generici method
                    && (cfg.WellKnownTypes.Delegate).IsAssignableFrom(m.Parameters[1])
                    && m.Parameters[2].IsEnum
                    && m.Parameters[3].Equals(cfg.WellKnownTypes.Boolean) == true
``` 

I would like to make a PR that makes it easier to identify the generic method. There are a few possibilities:

- Add `n to the end of name of generic method, where **n** is number of generic arguments
- Add property `IsGenericDef` to IXamlMethod 
- Expose GenericArguments and GenericParameters
- All three previous proposed solutions

What do you think is the best solution?

### PR DESCRIPTION
**Pull request by workgroupengineering**
Closes #112

### CODE REVIEW

File: tests/TypeSystemTest/BaseTest.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    {
+        _typeSystem = typeSystem;
+    }
+
+    protected IXamlTypeSystem TypeSystem => _typeSystem;
+
+    [Fact]
+    public void Generic_Types_Are_Correctly_Handled()
+    {
+        var wut = TypeSystem.GetType("TypeSystemTest.Models.TestType");
```
Comment (kekekeks) [inline tests/TypeSystemTest/BaseTest.cs]: As much as I like short variable names, they make the code way less readable when coming in such abundance. Can we have a bit more descriptive ones?

File: src/XamlX.IL.Cecil/CecilMethod.cs [hunk 2]
```diff
@@ -82,6 +82,27 @@ public CecilMethodBase(CecilTypeResolveContext typeResolveContext, MethodReferen
         [DebuggerDisplay("{" + nameof(Reference) + "}")]
         sealed class CecilMethod : CecilMethodBase, IXamlMethodBuilder<IXamlILEmitter>
         {
+            private IReadOnlyList<IXamlType> _genericParameters;
+            private IReadOnlyList<IXamlType> _genericArguments;
+
+            public bool IsGenericMethod => Reference.HasGenericParameters;
+            public bool IsGenericMethodDefinition => Reference.IsDefinition && Reference.HasGenericParameters;
+
+            public IReadOnlyList<IXamlType> GenericParameters => _genericParameters ??=
```
Comment (kekekeks) [inline src/XamlX.IL.Cecil/CecilMethod.cs:91]: I wonder how well would it handle scenarios like:
```cs
class Foo<TFoo>
{
    class Bar<TBar>
    {
         public void Test<TMethod>(TFoo foo, TBar bar, TMethod method) {}
    }
}
```

Comment (workgroupengineering) [reply src/XamlX.IL.Cecil/CecilMethod.cs:91]: Currently neither the Cecil nor Sre TypeSystem support nested classes.   For Cecil there is this PR #54 to enable support.

In Sre, I removed the following line for a quick test

https://github.com/kekekeks/XamlX/blob/941dafce490ffd21178f4493a7f0425e9d478e60/src/XamlX/IL/SreTypeSystem.cs#L131

after that, the following test is run that I expected

```c#
    [Fact]
    public void Nestest_Genercic_Class()
    {
        var nestedGenericTypeDefinition = TypeSystem.GetType("TypeSystemTest.Models.Foo`1+Bar`1");
        Assert.NotNull(nestedGenericTypeDefinition);

        Assert.Equal(2, nestedGenericTypeDefinition.GenericParameters.Count);

        var stringType = TypeSystem.GetType("System.String");
        var int32Type = TypeSystem.GetType("System.Int32");
        var boolType = TypeSystem.GetType("System.Boolean");

        var notGenericBarType = nestedGenericTypeDefinition.MakeGenericType(stringType, int32Type);

        var gm = notGenericBarType.Methods[0];

        Assert.NotNull(gm);
        Assert.True(gm.ContainsGenericParameters);
        Assert.Single(gm.GenericParameters);

        var im = gm.MakeGenericMethod([boolType]);
        Assert.NotNull(im);
        Assert.False(im.ContainsGenericParameters);
        Assert.Empty(im.GenericParameters);
    }
```

I think it is not the purpose of this PR to resolve the issue of nested classes.

File: tests/TypeSystemTest/BaseTest.cs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+{
+    private readonly IXamlTypeSystem _typeSystem;
+    private static readonly Dictionary<string, string> _cleanedFullName = [];
+
+    /// <summary>
+    /// See GitHub Issue https://github.com/kekekeks/XamlX/issues/119
+    /// </summary>
+    /// <param name="fullName"></param>
+    /// <returns></returns>
+    private static string GetNormalizzeFullName(string fullName)
```
Comment (maxkatz6) [inline tests/TypeSystemTest/BaseTest.cs]: nit: GetNormalizedFullName

**Review by maxkatz6 — APPROVED** [review]

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**